crates/circuit/src/interner.rs

```diff
@@ -343,7 +343,11 @@ where
     /// stored.
     ///
     /// This method does not store `value` if it is not present.
-    pub fn try_key(&self, value: &T) -> Option<Interned<T>> {
+    pub fn try_key<Q>(&self, value: &Q) -> Option<Interned<T>>
+    where
+        Q: ?Sized + Hash + Eq + ToOwned,
+        T: Borrow<Q> + ToOwned<Owned = <Q as ToOwned>::Owned>,
```

Comment:
Maybe we can instead reexport `indexmap`'s `Equivalent` with `pub use indexmap::Equivalent` in this file? And then just use it in these method bounds so it's easier to understand what's happening.

Reply:
I tried putting in `Q: Equivalent<T>` alone, but then the compiler will complain because, to satisfy the trait bounds of `Q: Equivalent<T>`, `Q` needs to be: `Q: ?Sized + ToOwned`, and `T` needs to satisfy `T: Borrow<Q>` as a minimum. The rest is more specificity to mention that the owned versions of both `T` and `Q` need to be of the same base type.

Bringing those trait bounds in already makes using `Q: Equivalent<T>` redundant, but it is implicitly satisfied. Is there an easier wat of conveying this?

Reply:
This works as long as we also specify that `T: Borrow<Q>`. Unfortunately, though, it's still not enough for the `insert` method.